### PR TITLE
fix(openapi): Return `Blob` instead of `Buffer` (matching external types)

### PR DESCRIPTION
**Pull request by davidkna-sap**
Related https://github.com/SAP/ai-sdk-js/pull/1923

### CODE REVIEW

File: packages/openapi/src/openapi-request-builder.ts [hunk 1]
```diff
@@ -339,6 +339,9 @@ export class OpenApiRequestBuilder<ResponseT = any> {
   ): Promise<ResponseT> {
     const response = await this.executeRaw(destination);
     if (isAxiosResponse(response)) {
+      if (response.data instanceof Buffer) {
+        return new Blob([response.data]) as ResponseT;
```
Comment (InjunPark-sap) [inline packages/openapi/src/openapi-request-builder.ts]:
```suggestion
        return new Blob([response.data], {
          type: response.headers?.['content-type'] ?? ''
        }) as ResponseT;
```

Comment (InjunPark-sap) [reply packages/openapi/src/openapi-request-builder.ts]: The `Blob` is created without a MIME type. We need to consider passing the `Content-Type` header.

**Review by InjunPark-sap — CHANGES_REQUESTED** [review]

**Review by marikaner — DISMISSED** [review]

**Comment by marikaner** [comment]
@InjunPark-sap, please re-review 😊

File: packages/openapi/src/openapi-request-builder.ts [hunk 2]
```diff
@@ -339,6 +339,16 @@ export class OpenApiRequestBuilder<ResponseT = any> {
...
     const response = await this.executeRaw(destination);
     if (isAxiosResponse(response)) {
+      if (Buffer.isBuffer(response.data)) {
+        const contentType = pickValueIgnoreCase(
+          response.headers,
+          'content-type'
+        );
+        return new Blob(
+          [response.data],
+          contentType && { type: contentType }
```
Comment (InjunPark-sap) [inline packages/openapi/src/openapi-request-builder.ts]:
```suggestion
          contentType ? { type: contentType } : undefined
```

File: packages/openapi/src/openapi-request-builder.spec.ts [hunk 3]
```diff
@@ -786,6 +786,83 @@ describe('openapi-request-builder', () => {
...
+      request: {}
+    } as any);
+    const result = await new OpenApiRequestBuilder<Blob>(
+      'get',
+      '/binary'
+    ).execute(destination);
+    expect(result).toBeInstanceOf(Blob);
+    expect(result.type).toBe('');
+  });
+
```
Comment (InjunPark-sap) [inline packages/openapi/src/openapi-request-builder.spec.ts:865]: Missing test case for empty string Content-Type. This is what would catch the bug above.

Suggestion: 
```
  it('does not throw when Content-Type is an empty string', async () => {
    const buffer = Buffer.from('binary content');
    httpSpy.mockResolvedValueOnce({
      data: buffer,
      status: 200,
      headers: { 'content-type': '' },
      config: {} as any,
      statusText: 'OK',
      request: {}
    } as any);
    const result = await new OpenApiRequestBuilder<Blob>('get', '/binary').execute(destination);
    expect(result).toBeInstanceOf(Blob);
    expect(result.type).toBe('');
  });
```

**Review by InjunPark-sap — CHANGES_REQUESTED** [review]

**Comment by InjunPark-sap** [comment]
@marikaner 
Thanks for the update — picking up `Content-Type` via `pickValueIgnoreCase` is actually an improvement over my original suggestion since it handles case-insensitive headers correctly.

One issue I figured out was introduced in the process: `contentType && { type: contentType }` passes an empty string `''` as the second argument to `new Blob()` when the server sends a `content-type: ''` header, which throws a `TypeError`. Please see the inline comments.

Feel free to give me feedback if it is right or not.

**Comment by marikaner** [comment]
> @marikaner Thanks for the update — picking up `Content-Type` via `pickValueIgnoreCase` is actually an improvement over my original suggestion since it handles case-insensitive headers correctly.
> 
> One issue I figured out was introduced in the process: `contentType && { type: contentType }` passes an empty string `''` as the second argument to `new Blob()` when the server sends a `content-type: ''` header, which throws a `TypeError`. Please see the inline comments.
> 
> Feel free to give me feedback if it is right or not.

good catch! seems to be fixed now, right?

**Comment by InjunPark-sap** [comment]
> > @marikaner Thanks for the update — picking up `Content-Type` via `pickValueIgnoreCase` is actually an improvement over my original suggestion since it handles case-insensitive headers correctly.
> > One issue I figured out was introduced in the process: `contentType && { type: contentType }` passes an empty string `''` as the second argument to `new Blob()` when the server sends a `content-type: ''` header, which throws a `TypeError`. Please see the inline comments.
> > Feel free to give me feedback if it is right or not.
> 
> good catch! seems to be fixed now, right?

Yes, all checks pass. Can we close it or is there anything to review?

**Comment by davidkna-sap** [comment]
@InjunPark-sap Please approve if the PR looks good to you or request changes if there are further issues.

**Review by InjunPark-sap — APPROVED** [review]

**Comment by InjunPark-sap** [comment]
@davidkna-sap Approved. Could we merge and close it despite of dependabot errror?
